index.js

```diff
@@ -26,7 +26,8 @@ class HttpProxy {
             agent: this.agent,
             protocol: config.protocol,
             hostname: config.hostname,
-            port: config.port
+            port: config.port,
+            timeout: config.timeout
```

Comment:
You added a timeout here but don't use it. What's this for?

Reply:
`timeout` is a part of `this.options` which is used later in:
```
            const options = Object.assign({}, this.options, {
                path: this.createProxyUrl(req),
                method: req.method
            });

            const proxyReq = this.http.request(options, (proxyRes) => {
                res.status(proxyRes.statusCode);
                this.copyResponseHeaders(
                    res,
                    this.filterResponseHeaders(req, proxyRes.headers)
                );
                proxyRes.pipe(res, {
                    end: true
                });
            });
```
so, that timeout sets request connection timeout and following code starts to work
```
const proxy = new HttpProxy({
    hostname: "example.com",
    protocol: "https:",
    timeout: 10000
});
```
README.md contains description about `timeout`, but without this line it didn't work.